frame/election-provider-support/src/lib.rs

```diff
@@ -318,8 +352,28 @@ pub trait ElectionDataProvider {
 	/// This is only useful for stateful election providers.
 	fn next_election_prediction(now: Self::BlockNumber) -> Self::BlockNumber;
 
+	/// Shorthand for "most significant page".
+	///
+	/// The least significant, aka "first" page that this election provider is expected to receive.
+	fn msp() -> PageIndex {
+		Self::Pages::get().defensive_saturating_sub(1)
+	}
+
+	/// Shorthand for "least significant page".
+	///
+	/// The most significant, aka "first" page that this election provider is expected to receive.
+	fn lsp() -> PageIndex {
+		0
+	}
+
+	/// The range of `take` pages, from most significant to least significant.
+	fn range(take: PageIndex) -> Box<dyn Iterator<Item = PageIndex>> {
```

Comment:
Why use trait object? Can we just return the concrete type and save the v table look up?